pkg/model/application_live_state.proto

```diff
@@ -71,6 +72,49 @@ message LambdaApplicationLiveState {
     repeated LambdaResourceState resources = 1;
 }
 
+message ApplicationLiveState {
+    repeated ResourceState resources = 1;
+
+    enum Status {
+        UNKNOWN = 0;
+        HEALTHY = 1;
+        OTHER = 2;
+    }
+    // The health status of the application.
+    Status health_status = 2;
+}
+
+// TODO: refine the actual state information
+// Existing fields: api_version, kind, namespace
+// Addtional fields: deploy_target
+message ResourceState {
+    enum HealthStatus {
+        UNKNOWN = 0;
+        HEALTHY = 1;
+        UNHEALTHY = 2;
+    }
+
+    // The unique ID.
+    string id = 1 [(validate.rules).string.min_len = 1];
+    // The sorted list of unique IDs of the parents.
+    repeated string parent_ids = 2;
+    // The name of resource.
+    string name = 3 [(validate.rules).string.min_len = 1];
+    // The type of this resource.
+    string resource_type = 4;
+    // The metadata of this resource.
+    map<string, string> resource_metadata = 5;
+    // The health status of this resource.
+    HealthStatus health_status = 6 [(validate.rules).enum.defined_only = true];
+    // The description of the health status.
+    string health_description = 7;
+
+    // The timestamp when this resource was created.
+    int64 created_at = 10 [(validate.rules).int64.gt = 0];
+    // The timestamp of the last time when this resource was updated.
+    int64 updated_at = 11 [(validate.rules).int64.gt = 0];
+}
```

Comment:
The diff compared to the existing livestate resource state for k8s, ecs, lambda, cloud run.

Deleted
- `owner_ids`: no longer used by any kinds of application.
- `api_version`, `kind`, `namespace`: these naming are k8s specific and we need to adopt the platform-neutral field for plugin architecture.

**Added**
- `resource_type`: We decide to use it as `kind` to show resource type.
- `resource_metadata`: We decided to add metadata as string map to add platform specific values freely (such as `api_version`, `namespace`)